engine/bsql_engine/io/io.pyx

```diff
@@ -243,9 +243,13 @@ cpdef parseMetadataCaller(fileList, offset, schema, file_format_hint, args):
     for i in range(names.size()):
         decoded_names.append(names[i].decode('utf-8'))
 
-    df = cudf.DataFrame(CudfXxTable.from_unique_ptr(blaz_move(dereference(resultSet).cudfTable), decoded_names)._data)
-    df._rename_columns(decoded_names)    
-    return df
+    dfs = []
+    for i in range(dereference(resultSet).cudfTables.size()):
+      df = cudf.DataFrame(CudfXxTable.from_unique_ptr(blaz_move(dereference(resultSet).cudfTables[i]), decoded_names)._data)
+      df._rename_columns(decoded_names)
+      dfs.append(df)
+
+    return dfs[0]
```

Comment:
We are expecting that parseMetadataPython only returns a single partition. We should add an assert somewhere here. We are implicitliy expecting that with this `return dfs[0]` we should add an assert to make it safer

Reply:
I removed that code. The modifications now are only for the runQuery function.